### DISCUSSION TITLE
Why isn't there a `wezterm-direct` option for `$TERM` env variable

### DISCUSSION THREAD

**Discussion by RayZ0rr** [wezterm/wezterm#4374] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

Herbstluftwm

### WezTerm version

wezterm 20230712-072601-f4abf8fd

### Ask your question!

I get the correct colourscheme when running any of the following to open [vifm](https://github.com/vifm/vifm):
```
TERM="xterm-direct" vf
TERM="tmux-direct" vf
TERM="alacritty-direct" vf
```
But with `TERM="wezterm" vf`  I get the same result as `TERM="xterm-256color" vf`

**Comment by wez** [comment, marked as answer]
no one has needed one before.
You can safely use `xterm-direct` if you wish; wezterm is intentionally built to be compatible with xterm so that you shouldn't need to provide custom terminfo entries.

**Comment by vollink** [comment]
Because every vendor that offers an `xterm` definition caters to the lowest common denominator (since, well, every terminal emulator claims to be xterm compatible), I _also_ am a stickler for using the CORRECT terminal definition when possible.  Thing is, TIC is actually really easy to deal with.  If you (OP) are already downloading the official wezterm.terminfo, you can just copy it to your own wezterm-direct.terminfo, find `setaf` and `setab` and put in the xterm+direct2 definition...

```
        setab=\E[%?%p1%{8}%<%t4%p1%d%e48:2:%p1%{65536}%/%d:%p1%{256}
            %/%{255}%&%d:%p1%{255}%&%d%;m,
        setaf=\E[%?%p1%{8}%<%t3%p1%d%e38:2:%p1%{65536}%/%d:%p1%{256}
            %/%{255}%&%d:%p1%{255}%&%d%;m,
```
Remember to change the name at the top of the file and `tic -x your-new-filename`  Poof, you've got your file for `TERM=wezterm-direct`

EDIT -- This is the version I tested to be working for me on my own wezterm.  (Thanks for the idea)

**Reply by vollink** [reply]
This one is the cleanest way I've found to create a `wezterm-direct`.  It does require that the official wezterm.terminfo should be compiled and available already, but this file can be run through tic directly without all the other stuff...

file: `wezterm-direct.terminfo`:
```terminfo
# wezterm-direct settings from xterm+direct2 entry of:
# https://ncurses.scripts.mit.edu/?p=ncurses.git;a=blob_plain;f=misc/terminfo.src;hb=HEAD
wezterm-direct|Wez's terminal emulator (direct color),
    RGB,
    colors#0x1000000, pairs#0x10000, CO#8,
    setab=\E[%?%p1%{8}%<%t4%p1%d%e48:2:%p1%{65536}%/%d:%p1%{256}
        %/%{255}%&%d:%p1%{255}%&%d%;m,
    setaf=\E[%?%p1%{8}%<%t3%p1%d%e38:2:%p1%{65536}%/%d:%p1%{256}
        %/%{255}%&%d:%p1%{255}%&%d%;m,
    use=wezterm,
```
`$ tic -x wezterm-direct.terminfo`

Technically, the same code block above could be added directly into the wezterm.terminfo itself, where `tic -x ` would automatically output both definitions.

PLEASE KNOW that older versions of tic will have issue with the COLORS and PAIRS counts and will issue warnings about relegating them to smaller values.  That is a limitation of the older terminfo versions, but otherwise doesn't harm the capabilities of the `setab` and `setaf` entries.
